kernel/syscalls.c

```diff
@@ -152,6 +153,75 @@ ulong sys_times(struct tms *user_buf)
    return (ulong) get_ticks();
 }
 
+static struct k_timeval 
+timespec_to_timeval(struct k_timespec64 ts)
```

Comment:
I didn't find an equivalent function. Should I move it to tilck/kernel/datetime.h?

Reply:
Yeah, this should go to `tilck/kernel/datetime.h` but we should be very careful about its name as there are several timespec and timeval structs. I'd suggest calling it `k_ts64_to_k_timeval()`  or `k_timespec_to_k_timeval()`, implying that a timespec is 64-bit by default. Something like that. The `k_` prefix matters to avoid namespace-clashing with the userspace structs, which are also included in the headers. Including those headers in the kernel is both good and bad. Good because we're implementing Linux compatibility at binary level anyway, so why re-defining everything? Also, including those headers forces us to use non-standard names when there is a clashing and that is good to avoid causing confusion and make searching of identifiers easier. But including the system headers is also bad because sometimes the userspace structs are different from the kernel-space ones and because some userspace structs have extra hidden (non-officially documented)  `__fields` depending on the libc (glibc vs musl for example). So, for all of those cases, we need to have a dedicated `k_` struct in the kernel. 

Just to share the whole story with you, I even developed a fancy plugin for the clang static analyzer that allows me to check if any struct in Tilck changed its binary layout when compiling the kernel using a different system root for the system headers. I haven't been using it for 2-3 years, but it was very useful to check that the Tilck doesn't expose to userspace structs that are different, depending on how it is compiled. We might need it again in the future.

Reply:
Right! I'll go with `k_ts64_to_k_timeval`. Thank you for the explanation, the whole thing wasn't very clear to me.